src/merkle/store.rs

```diff
@@ -346,49 +365,145 @@ mod test {
     ];
 
     #[test]
-    fn test_add_merkle_tree() -> Result<(), MerkleError> {
+    fn test_root_not_in_storage() -> Result<(), MerkleError> {
+        let mtree = MerkleTree::new(LEAVES4.to_vec())?;
+        let store = MerkleStore::default().with_merkle_tree(LEAVES4)?;
+        assert_eq!(
+            store.get_node(LEAVES4[0], NodeIndex::new(mtree.depth(), 0)),
+            Err(MerkleError::RootNotInStorage(LEAVES4[0])),
+            "Leaf 0 is not a root"
+        );
+        assert_eq!(
+            store.get_path(LEAVES4[0], NodeIndex::new(mtree.depth(), 0)),
+            Err(MerkleError::RootNotInStorage(LEAVES4[0])),
+            "Leaf 0 is not a root"
+        );
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_merkle_tree() -> Result<(), MerkleError> {
         let mut store = MerkleStore::default();
 
         let mtree = MerkleTree::new(LEAVES4.to_vec())?;
         store.add_merkle_tree(LEAVES4.to_vec())?;
 
-        assert!(
+        // STORE LEAVES ARE CORRECT ==============================================================
+        // checks the leaves in the store corresponds to the expected values
+        assert_eq!(
             store
                 .get_node(mtree.root(), NodeIndex::new(mtree.depth(), 0))
-                .is_ok(),
+                .unwrap(),
+            LEAVES4[0],
             "node 0 must be in the tree"
         );
```

Comment:
This formatting makes it a bit difficult to follow. Maybe update to something like:
```Rust
assert_eq!(
    store.get_node(mtree.root(), NodeIndex::new(mtree.depth(), 0)),
    Ok(LEAVES4[0]),
    "node 0 must be in the tree"
);
```
If this doesn't work, maybe introduce `depth` variable (set to `mtree_depth()`) to make `rustfmt` accept it?